cpu/esp32: enable BLE and NimBLE host support for ESP32-C3

gschorcht opened the pull request:
### Contribution description

This PR enables the Bluetooth 5 including advertising extension and NimBLE host support for ESP32-C3.

BLE and NimBLE host support for ESP32-C3 including advertising extension was already tested when BLE and NimBLE host support was tesed for ESP32 in  PR #18439.

### Testing procedure

The easiest way to test the PR is to use an ESP32 node and an ESP32-C3 node and flash `tests/nimble_autoconn_gnrc` and test it.
```
BOARD=esp32-wroom-32 make -j8 -C examples/nimble_autoconn_gnrc flash term
```
```
BOARD=esp32c3-devkit make -j8 -C examples/nimble_autoconn_gnrc flash term
```
Use command `ble info` to check that a connection between the two nodes is established after some seconds.

Node 1:
```
> ble info
Own Address: 60:55:F9:6E:6E:69 (public) -> [FE80::6055:F9FF:FE6E:6E69]
Supported PHY modes: 1M
 Free slots: 1/3
Advertising: yes
Connections: 1
[ 0] 7C:DF:A1:E2:94:99 [FE80::7CDF:A1FF:FEE2:9499] (M,75ms,2500ms,0,1M)
     (role, conn itvl, superv. timeout, slave latency, PHY)
Slots:
[ 0] state: 0x0011 - GAP-master L2CAP-client
[ 1] state: 0x0100 - advertising
[ 2] state: 0x8000 - unused
```
Node 2:
```
> ble info
Own Address: 7C:DF:A1:E2:94:99 (public) -> [FE80::7CDF:A1FF:FEE2:9499]
Supported PHY modes: 1M
 Free slots: 2/3
Advertising: no
Connections: 1
[ 0] 60:55:F9:6E:6E:69 [FE80::6055:F9FF:FE6E:6E69] (S,75ms,2500ms,0,1M)
     (role, conn itvl, superv. timeout, slave latency, PHY)
Slots:
[ 0] state: 0x0022 - GAP-slave L2CAP-server
[ 1] state: 0x8000 - unused
[ 2] state: 0x8000 - unused
```

### Issues/PRs references

benpicco commented on cpu/esp32/Makefile.include:
```diff
@@ -164,7 +164,11 @@ LINKFLAGS += -L$(RIOTCPU)/$(CPU)/ld/$(CPU_FAM)/
 ifeq (,$(filter esp_ble,$(USEMODULE)))
```
If only `esp_ble_esp32` requires the special linker script, why not
```suggestion
ifeq (,$(filter esp_ble_esp32,$(USEMODULE)))
```

gschorcht replied:
`esp_ble_esp32` is only a feature, but we could use
```
ifeq (,$(filter esp32_sdk_lib_bt_esp32,$(USEPKG)))
```
instead.

benpicco replied:
I thought features get turned into modules at the end of dependency resolution, but that will also work.

benpicco reviewed (approved):
Looking forward to ESP32-S3 BLE support now :smiley:

gschorcht commented:
@benpicco May I squash?

benpicco commented:
Of course :)

benpicco commented:
KConfig doesn't match :confused:

gschorcht commented:
> KConfig doesn't match confused

Hm, I tried the feature immediatily before I left for vacation and forgot to enable it in Kconfig.

gschorcht commented:
> > KConfig doesn't match confused
> 
> Hm, I tried the feature immediatily before I left for vacation and forgot to enable it in Kconfig.

I added `select HAS_BLE_PHY_2MBIT` to `cpu/esp32/Kconfig.esp32c3`. I squashed it directly because it was a marginal solution.

gschorcht commented:
> Looking forward to ESP32-S3 BLE support now smiley

See PR #18517.

gschorcht commented:
@benpicco Thanks for reviewing and merging.
